budgetportal/models.py

```diff
@@ -768,8 +769,324 @@ def get_expenditure_over_time(self):
+                                 'name': cell[econ_class_3_ref],
+                                 'amount': cell['value.sum']}
+            if cell[econ_class_2_ref] not in econ_classes.keys():
+                if cell['value.sum'] != 0:
+                    econ_classes[cell[econ_class_2_ref]] = dict()
+                    econ_classes[cell[econ_class_2_ref]]['type'] = 'economic_classification_2'
+                    econ_classes[cell[econ_class_2_ref]]['name'] = cell[econ_class_2_ref]
+                    econ_classes[cell[econ_class_2_ref]]['items'] = [new_econ_2_object]
+            else:
+                if cell['value.sum'] != 0:
+                    econ_classes[cell[econ_class_2_ref]]['items'].append(new_econ_2_object)
+        # sort by name
+        name_func = lambda x: x['name']
+        for econ_2_name in list(econ_classes.keys()): # Copy keys because we're updating dict
+            econ_classes[econ_2_name]['items'] = sorted(
+                econ_classes[econ_2_name]['items'], key=name_func)
+        econ_classes_list = sorted(econ_classes.values(), key=name_func)
+        return econ_classes_list if econ_classes_list else None
+
+    @staticmethod
+    def _get_total_budget_adjustment(openspending_api, cells):
+        if not cells:
+            return None
+        phase_ref = openspending_api.get_phase_ref()
+        descript_ref = openspending_api.get_adjustment_kind_ref()
+        total_adjusted, total_voted = None, None
+
+        for cell in cells:
+            if cell[phase_ref] == 'Adjusted appropriation' and \
+                    cell[descript_ref] == 'Adjustments - Total adjustments':
+                total_adjusted = cell['value.sum']
+            if cell[phase_ref] == 'Voted (Main appropriation)' and \
+                    cell[descript_ref] == 'Total':
+                total_voted = cell['value.sum']
+
+        if total_voted and not total_adjusted:
+            total_adjusted = 0
+        elif not (total_voted or total_adjusted):
+            raise Exception("Could not calculate total change for department budget")
+
+        return total_voted, total_adjusted
+
+    def _get_budget_virements(self, openspending_api, dataset, total_voted):
+        virements_resource = dataset.get_resource('CSV', name='Value of Virements')
+        if virements_resource:
+            sql = '''
+                    SELECT "Value of Virements" FROM "{}"
+                    WHERE "department_name"='{}'
+                    '''.format(virements_resource['id'], self.name)
+            params = {
+                'sql': sql
+            }
+            result = requests.get(CKAN_DATASTORE_URL, params=params)
+            result.raise_for_status()
+            records = result.json()['result']['records']
+            if records:
+                value = records[0]['Value of Virements']
+            else:
+                raise Exception('Value of Virements query returned no records')
```

Comment:
you could even have let this just be an index out of range error and skipped the condition, since we're not handling it in a special way - the exception would lead us right to the problem.

Reply:
Good point, thanks.